core/frontend/src/tile/RealityModelTileTree.ts

```diff
@@ -452,6 +452,53 @@ export namespace RealityModelTileTree {
     const tileClient = new RealityModelTileClient(url, accessToken, iModel.contextId);
     const json = await tileClient.getRootDocument(url);
     let rootTransform = iModel.ecefLocation ? iModel.backgroundMapLocation.getMapEcefToDb(0) : Transform.createIdentity();
+    const geoConverter = iModel.noGcsDefined ? undefined : iModel.geoServices.getConverter("WGS84");
+    if (geoConverter !== undefined) {
+      let realityTileRange = RealityModelTileUtils.rangeFromBoundingVolume(json.root.boundingVolume)!.range;
+      if (json.root.transform) {
+        const realityToEcef = RealityModelTileUtils.transformFromJson(json.root.transform);
+        realityTileRange = realityToEcef.multiplyRange(realityTileRange);
+      }
+
+      const ecefOrigin = realityTileRange.localXYZToWorld(.5, .5, .5)!;
+
+      const ecefEastPoint = ecefOrigin.plusXYZ(10, 0, 0);
+      const ecefNorthPoint = ecefOrigin.plusXYZ(0, 10, 0);
+      const ecefElevationPoint = ecefOrigin.plusXYZ(0, 0, 10);
+
+      const cartographicOrigin = Cartographic.fromEcef(ecefOrigin)!;
+      const cartographicNorth = Cartographic.fromEcef(ecefNorthPoint)!;
+      const cartographicEast = Cartographic.fromEcef(ecefEastPoint)!;
+      const cartographicElevation = Cartographic.fromEcef(ecefElevationPoint);
+
+      const geoOrigin = Point3d.create(cartographicOrigin.longitudeDegrees, cartographicOrigin.latitudeDegrees, cartographicOrigin.height);
+      const geoNorth = Point3d.create(cartographicNorth.longitudeDegrees, cartographicNorth.latitudeDegrees, cartographicNorth.height);
+      const geoEast = Point3d.create(cartographicEast.longitudeDegrees, cartographicEast.latitudeDegrees, cartographicEast.height);
+      const geoElevation = Point3d.create(cartographicElevation?.longitudeDegrees, cartographicElevation?.latitudeDegrees, cartographicElevation?.height);
+
+      const response = await geoConverter.getIModelCoordinatesFromGeoCoordinates([geoOrigin, geoNorth, geoEast, geoElevation]);
+      if (response.iModelCoords[0].s === GeoCoordStatus.Success && response.iModelCoords[1].s === GeoCoordStatus.Success && response.iModelCoords[2].s === GeoCoordStatus.Success && response.iModelCoords[3].s === GeoCoordStatus.Success) {
+        const iModelOrigin = Point3d.fromJSON(response.iModelCoords[0].p);
+        const iModelNorth = Point3d.fromJSON(response.iModelCoords[1].p);
+        const iModelEast = Point3d.fromJSON(response.iModelCoords[2].p);
+        const iModelElevation = Point3d.fromJSON(response.iModelCoords[3].p);
+
+        const ecefXVector = Vector3d.createStartEnd(ecefOrigin, ecefEastPoint);
+        const ecefYVector = Vector3d.createStartEnd(ecefOrigin, ecefNorthPoint);
+        const ecefZVector = Vector3d.createStartEnd(ecefOrigin, ecefElevationPoint);
+        const frameA = Transform.createOriginAndMatrixColumns(ecefOrigin, ecefXVector, ecefYVector, ecefZVector);

```

Comment:
No - this is wrong -- You need to test for modelZVector being 0 magnitude.   ecefXVector, ecefYVector and ecefZVector are always just going to be (10, 0, 0), (0, 10, 0) and (0, 0, 10).   You should create variables for these rather than recalculationg.